core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -283,57 +251,154 @@ public Options saveKeyValues() {
     }
 
     @Override
-    public Options overTabletRange() {
+    public RangeOptions scanTable(String tableName) {
+      this.table = tableName;
       this.range = TabletsSection.getRange();
       return this;
     }
+  }
 
-    @Override
-    public Options overRange(Range range) {
-      this.range = range;
-      return this;
+  public interface Options {
+    TabletsMetadata build(AccumuloClient client);
+
+    TabletsMetadata build(ClientContext ctx);
+
+    /**
+     * Checks that the metadata table forms a linked list and automatically backs up until it does.
+     */
+    Options checkConsistency();
+
+    Options fetchCloned();
+
+    Options fetchCompactId();
+
+    Options fetchDir();
+
+    Options fetchFiles();
+
+    Options fetchFlushId();
+
+    Options fetchLast();
+
```

Comment:
Could add javadoc to all of these options.

Reply:
I think it would be better to document the metadata table schema in one place and have this class refer to that.